pkg/cluster/provider.go

```diff
@@ -58,13 +59,22 @@ func NewProvider(options ...ProviderOption) *Provider {
 	for _, o := range options {
 		o.apply(p)
 	}
-	if p.provider == nil {
-		// check if provider was overridden
-		// TODO: consider auto-detection once more than 1 provider is stable
-		providerEnv := os.Getenv("KIND_EXPERIMENTAL_PROVIDER")
-		if providerEnv == "podman" {
+
+	supportedProviders := map[string]func(log.Logger) internalprovider.Provider{
+		"podman": podman.NewProvider,
+		"docker": docker.NewProvider,
+	}
+
+	// check if provider was overridden
```

Comment:
shouldn't this only be checked if `p.provider == nil`?

Reply:
yes, but currently at this point `p.provider` is always `nil`.

that being said I will closely be sending a follow up PR that will expose the provider as a ProviderOption, and probably also move the `KIND_EXPERIMENTAL_PROVIDER` up to the cli.

Reply:
I agree that we should check for nil but also think that follow up should happen